### PR TITLE
Migrate AWS Gems to CDKv2 

### PR DESCRIPTION
**Pull request by lmbr-pip**
## What does this PR do?

The Cloudformation Development Kit (CDK) has a new long-term stable version CDKv2. The prior CDK major version, CDK v1 entered maintenance on June 1, 2022 and will now receive only critical bug fixes and security patches.

This PR consists of:
* CDKv2 updates for AWSMetrics, AWSCore, AWSClientAuth and AWSGameLift - all CDKs have been synthed and deployed
* Updates to the automation tests

Prior to merging PR should perform these actions:
- [x] Refresh from Development
- [x] Re-run AWS based integration tests
- [x] Run local periodic tests to confirm setup
- [x] Confirm the correct settings for Jenkins files
- [x] Breaking change notification, legacy users will have to use the cdkv1 stacks 
- [x] Docs PR approved that covers change to CDKv2 and links to migration guide. See https://github.com/o3de/o3de.org/pull/1950

## How was this PR tested?
Re-ran the automation tests locally using CDKV2  ```Set CDK_VERSION=2.49.0```

```
    python\python.cmd -m pytest E:\Dev\o3de\o3de\AutomatedTesting\Gem\PythonTests\AWS\core\test_aws_resource_interaction.py --build-directory E:\dev\o3de\o3de\build\windows_vs2019\bin\profile
	
	python\python.cmd -m pytest E:\Dev\o3de\o3de\AutomatedTesting\Gem\PythonTests\AWS\aws_metrics\aws_metrics_automation_test.py --build-directory E:\dev\o3de\o3de\build\windows_vs2019\bin\profile
	
	python\python.cmd -m pytest E:\Dev\o3de\o3de\AutomatedTesting\Gem\PythonTests\AWS\client_auth\aws_client_auth_automation_test.py --build-directory E:\dev\o3de\o3de\build\windows_vs2019\bin\profile
```
> Had to explicit turn off the EC2 metadata credential search as this led to timeouts in the test.

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 1]
```diff
@@ -3,12 +3,16 @@
 ## Prerequisites
 1. Build the O3DE Editor and AutomatedTesting.GameLauncher in Profile.
 2. Install the latest version of NodeJs.
-3. AWS CLI is installed and AWS crendentials are configured via [environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) or [default profile](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html).
-4. [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html#getting_started_install) is installed.
+3. AWS CLI is installed and AWS credentials are configured via [environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) or [default profile](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html).
+4. [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_install) is installed.
+
+> Note: ALl the AWS Gems CDK applications have been updated to CDKv2. Ensure you set CDK_VERSION to the desired CDKv2 version. See https://docs.aws.amazon.com/cdk/api/versions.html for the latest CDKv2 version.
```
Comment (allisaurus) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:9]: nit: casing weirdness 
> ALl  --> All

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 2]
```diff
@@ -20,17 +24,19 @@
...
         ```
-            Set O3DE_AWS_PROJECT_NAME=AWSAUTO
-            Set O3DE_AWS_DEPLOY_REGION=us-east-1
-            Set ASSUME_ROLE_ARN=arn:aws:iam::{your_aws_account_id}:role/o3de-automation-tests
-            Set COMMIT_ID=HEAD
+            set O3DE_AWS_PROJECT_NAME=AWSAUTO
+            set O3DE_AWS_DEPLOY_REGION=us-east-1
+            set ASSUME_ROLE_ARN=arn:aws:iam::{your_aws_account_id}:role/o3de-automation-tests
+            set COMMIT_ID=HEAD
+            set CDK_VERSION=2.39.1
```
Comment (allisaurus) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:31]: will the tests fail on another version of the CDK v2?

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 3]
```diff
@@ -46,8 +52,16 @@
             python/python.sh -m pytest {path_to_the_test_file} --build-directory {directory_to_the_profile_build}
         ```
 
-### Pycharm
-You can also run any specific automation test directly from Pycharm by providing the "--build-directory" argument in the Run Configuration.
+The main tests files can be found under this directory at the following paths:
```
Comment (allisaurus) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:55]: thank you for adding this! very handy

File: AutomatedTesting/Gem/PythonTests/AWS/README.md (hunk 2 above)
Comment (lmbr-pip) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md:31]: No, was unclear how to provide an example, this is latest version

**Comment by lmbr-pip** [comment]
Docs changes for this: https://github.com/o3de/o3de.org/pull/1950

File: AutomatedTesting/Gem/PythonTests/AWS/README.md (hunk 3 above)
Comment (lmbr-pip) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md:55]: It slowed me down trying to understand where these files so though it had value.

File: AutomatedTesting/Gem/PythonTests/AWS/README.md (hunk 1 above)
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:9]:
```suggestion
> Note: All the AWS Gems CDK applications have been updated to CDK v2. Ensure you set CDK_VERSION to the desired CDK v2 version. See https://docs.aws.amazon.com/cdk/api/versions.html for the latest CDK v2 version information.

```

Comment (lmbr-pip) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md]: Fixed

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 4]
```diff
@@ -20,17 +25,19 @@
...
         ```
-            Set O3DE_AWS_PROJECT_NAME=AWSAUTO
-            Set O3DE_AWS_DEPLOY_REGION=us-east-1
-            Set ASSUME_ROLE_ARN=arn:aws:iam::{your_aws_account_id}:role/o3de-automation-tests
-            Set COMMIT_ID=HEAD
+            set O3DE_AWS_PROJECT_NAME=AWSAUTO
+            set O3DE_AWS_DEPLOY_REGION=us-east-1
+            set ASSUME_ROLE_ARN=arn:aws:iam::{your_aws_account_id}:role/o3de-automation-tests
+            set COMMIT_ID=HEAD
+            set CDK_VERSION=2.39.1
```
Comment (junbo75) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:32]: I think we need to update the CDK version in these files as well to run the tests on Jenkins with CDK v2: 

- https://github.com/o3de/o3de/blob/development/scripts/build/Platform/Windows/build_config.json#L507
- https://github.com/o3de/o3de/blob/development/scripts/build/Platform/Linux/build_config.json#L381

File: AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_utils.py [hunk 5]
```diff
@@ -224,6 +224,22 @@ def delete_table(self, database_name: str, table_name: str) -> None:
             Name=table_name
         )
 
+    def try_delete_table(self, database_name: str, table_name: str) -> None:
```
Comment (junbo75) [inline AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_utils.py:227]: I think we can get rid of the delete_table function if it's not used by the test code any more:

https://github.com/o3de/o3de/blob/development/AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_utils.py#L214

**Review by junbo75 — APPROVED** [review]
Looks good to me.

File: AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_utils.py (hunk 5 above)
Comment (lmbr-pip) [reply AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_utils.py:227]: Hmm, good call, will see its still used and remove if its now been abandoned.

File: AutomatedTesting/Gem/PythonTests/AWS/README.md (hunk 4 above)
Comment (lmbr-pip) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md:32]: Good call, Will update those.

File: AutomatedTesting/Gem/PythonTests/AWS/README.md (hunk 2 above)
Comment (allisaurus) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md:32]: what happens if this isn't set at all, will it attempt to use the latest CDK v2? I understand for automation testing it's useful to be explicit about what version to use but requiring it adds a bit more friction for local users

**Review by allisaurus — APPROVED** [review]

**Comment by lmbr-pip** [comment]
Will return to this once the 2210 release is out to avoid any disruption.

**Comment by lmbr-pip** [comment]
Will be recommence work week of Oct 31st. Work is to address https://github.com/o3de/o3de/issues/12848

**Review by junbo75 — COMMENTED** [review]
Changes look good to me. Do we want to add the AWS_EC2_METADATA_DISABLED configuration to the Jenkins environment variable as well so that it can be changed on Jenkins?

Please check https://github.com/o3de/o3de/blob/development/scripts/build/Platform/Windows/pipeline.json and https://github.com/o3de/o3de/blob/development/scripts/build/Platform/Linux/pipeline.json for setting Jenkins variables.

**Review by junbo75 — COMMENTED** [review]
Maybe update the CDK version we use on Jenkins as well at:

**Review by allisaurus — APPROVED** [review]
LGTM, though its strange that using named args in `_setup(...)` fixed the integ tests 🤔

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 6]
```diff
@@ -26,7 +26,7 @@
     * Windows
         ```
             set O3DE_AWS_PROJECT_NAME=AWSAUTO
-            set O3DE_AWS_DEPLOY_REGION=us-east-1
+			set O3DE_AWS_DEPLOY_REGION=us-east-1
```
[the remark this answers is not in the document]
Comment (lmbr-pip) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md:29]: Yep, its probably tabs being converted to spaces. Can fix

**Pull request merged**